### PR TITLE
optimizer: Fix a bug in typeinferer

### PR DESCRIPTION
**Pull request by zxylvlp**

### CODE REVIEW

File: optimizer/typeinferer.go [hunk 1]
```diff
@@ -258,6 +258,9 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 	switch x.FnName.L {
 	case "abs", "ifnull", "nullif":
 		tp = x.Args[0].GetType()
+		if x.FnName.L == "abs" && tp.Tp == mysql.TypeDatetime {
```
Comment (shenli) [inline optimizer/typeinferer.go]: Consider other types?

**Comment by coocood** [comment]
LGTM

**Comment by zxylvlp** [comment]
@shenli @zimulala PTAL

**Comment by shenli** [comment]
LGTM

**Pull request merged**